airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -385,175 +383,296 @@ static Optional<ContainerOrchestratorConfig> getContainerOrchestratorConfig(fina
-    final StreamResetPersistence streamResetPersistence = new StreamResetPersistence(configDatabase);
-
-    final TemporalClient temporalClient = new TemporalClient(workflowClient, configs.getWorkspaceRoot(), temporalService, streamResetPersistence);
-    TemporalUtils.configureTemporalNamespace(temporalService);
+  private static void initializeCommonDependencies() {
+    LOGGER.info("Initializing common worker dependencies.");
+    configs = new EnvConfigs();
+    LOGGER.info("workspaceRoot = " + configs.getWorkspaceRoot());
 
-    final TemporalWorkerRunFactory temporalWorkerRunFactory = new TemporalWorkerRunFactory(
-        temporalClient,
-        workspaceRoot,
-        configs.getAirbyteVersionOrWarning(),
-        featureFlags);
+    MetricClientFactory.initialize(MetricEmittingApps.WORKER);
 
-    final WorkspaceHelper workspaceHelper = new WorkspaceHelper(
-        configRepository,
-        jobPersistence);
+    LogClientSingleton.getInstance().setWorkspaceMdc(
+        configs.getWorkerEnvironment(),
+        configs.getLogConfigs(),
+        LogClientSingleton.getInstance().getSchedulerLogsRoot(configs.getWorkspaceRoot()));
 
-    final ConnectionHelper connectionHelper = new ConnectionHelper(configRepository, workspaceHelper);
+    if (configs.getWorkerEnvironment().equals(WorkerEnvironment.KUBERNETES)) {
+      KubePortManagerSingleton.init(configs.getTemporalWorkerPorts());
+    }
 
-    final Optional<ContainerOrchestratorConfig> containerOrchestratorConfig = getContainerOrchestratorConfig(configs);
+    featureFlags = new EnvVariableFeatureFlags();
+    defaultWorkerConfigs = new WorkerConfigs(configs);
+    temporalService = TemporalUtils.createTemporalService();
+    workflowClient = TemporalUtils.createWorkflowClient(temporalService, TemporalUtils.getNamespace());
+    TemporalUtils.configureTemporalNamespace(temporalService);
+    airbyteApiClient = getApiClient(configs);
+  }
 
-    final WebUrlHelper webUrlHelper = new WebUrlHelper(configs.getWebappUrl());
+  private static void initializeControlPlaneDependencies() throws IOException, DatabaseCheckException {
+    if (!configs.isControlPlaneWorker()) {
+      LOGGER.info("Skipping Control Plane dependency initialization.");
+      return;
+    }
+    LOGGER.info("Initializing control plane worker dependencies.");
+
+    final DataSource configsDataSource = DataSourceFactory.create(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(),
+        DRIVER_CLASS_NAME, configs.getConfigDatabaseUrl());
+    final DataSource jobsDataSource = DataSourceFactory.create(configs.getDatabaseUser(), configs.getDatabasePassword(),
+        DRIVER_CLASS_NAME, configs.getDatabaseUrl());
+
+    // Manual configuration that will be replaced by Dependency Injection in the future
+    try (final DSLContext configsDslContext = DSLContextFactory.create(configsDataSource, SQLDialect.POSTGRES);
```

Comment:
@evantahler @davinchia parent PR discussion:
![image](https://user-images.githubusercontent.com/7798112/186284841-e5a47907-78e7-4344-b1ac-c3a848a085c5.png)

Reply:
Would it make sense to abstract this code into a function? Is this the same thing we do in other places? The confusion about migrations seems justified.

Reply:
I'd prefer to leave that outside the scope of this PR, since I don't have any context on how this works and it isn't directly related to multi-cloud. The only reason it's showing up in the diff of this PR is because I moved it from the main method to a helper for readability, it used to be here: https://github.com/airbytehq/airbyte/pull/15904/files#diff-c87125243174e26855509edab4318ed66dd21a7a90204a10a40e86a3b6ed1b05L531

Reply:
okay. please add a todo comment then. happy to punt on it as long as we leave a reminder for ourselves.